backend/tracim_backend/lib/core/event.py

```diff
@@ -1055,3 +1057,53 @@ def context(self) -> Generator[TracimContext, None, None]:
     def publish_messages_for_event(self, event_id: int) -> None:
         logger.debug(self, "publish event(id={}) synchronously".format(event_id))
         self._publish_messages_for_event(event_id)
+
+
+class MessageHooks:
+    def _create_workspace_historic_messages_for_current_user(
+        self, role: UserRoleInWorkspace, context: TracimContext
+    ):
+        current_user = context.safe_current_user()
+        event_api = EventApi(current_user, context.dbsession, context.app_config)
+        event_api.create_messages_history_for_user(
+            user_id=role.user_id,
+            workspace_ids=[role.workspace_id],
+            max_messages_count=context.app_config.WORKSPACE__JOIN__MAX_MESSAGES_HISTORY_COUNT,
+        )
+
+    @hookimpl
+    def on_user_role_in_workspace_created(
+        self, role: UserRoleInWorkspace, context: TracimContext
+    ) -> None:
+        self._create_workspace_historic_messages_for_current_user(role, context)
+
+    @hookimpl
+    def on_workspace_deleted(self, workspace: Workspace, context: TracimContext) -> None:
+        current_user = context.safe_current_user()
+        event_api = EventApi(current_user, context.dbsession, context.app_config)
+        event_api.delete_message_for_workspace(workspace.workspace_id)
+
+    @hookimpl
+    def on_workspace_modified(self, workspace: Workspace, context: TracimContext) -> None:
+        current_user = context.safe_current_user()
+        event_builder = EventBuilder(context.app_config)
+        event_api = EventApi(current_user, context.dbsession, context.app_config)
+        if event_builder._has_just_been_deleted(workspace):
```

Comment:
note: a refactor to be easily able to reuse `_has_just_been_deleted` / `_has_just_been_undeleted`  without calling internal property should be done. don't think much about it for now.